ic-canister/ic-canister/src/lib.rs

```diff
@@ -452,8 +452,14 @@ pub fn call_virtual_responder(
             None => Err((
                 RejectionCode::Unknown,
                 format!(
-                    "canister method {} is not registered for principal {}",
-                    method_name, principal
+                    "canister method {} is not registered for principal {}, METHODS: {:?}",
+                    method_name,
+                    principal,
+                    responders
+                        .borrow()
+                        .keys()
+                        .map(|(k, r)| (k.to_string(), r))
+                        .collect::<Vec<_>>()
                 ),
```

Comment:
You can move the `method_name` and `principal` directly into the string it self:

```rust
                format!(
                    "canister method {method_name} is not registered for principal {principal}, METHODS: {:?}",
                    responders
                        .borrow()
                        .keys()
                        .map(|(k, r)| (k.to_string(), r))
                        .collect::<Vec<_>>()
                ),
```